roofit/roofitcore/test/testRooFuncWrapper.cxx

```diff
@@ -476,7 +478,43 @@ FactoryTestParams param5{"SimPdf", getSimPdfModel,
                          5e-3,
                          /*randomizeParameters=*/true};
 
-INSTANTIATE_TEST_SUITE_P(RooFuncWrapper, FactoryTest, testing::Values(param1, param2, param3, param4, param5),
+namespace {
+void getDataHistModel(RooWorkspace &ws)
+{
+   RooRealVar x("x", "x", 6, 0, 20);
+   RooPolynomial p("p", "p", x, RooArgList(0.01, -0.01, 0.0004));
+
+   // Sample 500 events from p
+   x.setBins(10);
+   RooDataSet *data1 = p.generate(x, 500);
+
+   // Create a binned dataset with 20 bins and 500 events
+   RooDataHist *hist1 = data1->binnedClone();
```

Comment:
```suggestion
   std::unique_ptr<RooDataHist> hist1{p.generateBinned(x, 500)};
```
Memory leak.
You can use `generateBinned()`, then you don't need the intermediate RooDataSet.